modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/user/UserManager.kt

```diff
@@ -114,8 +114,8 @@ class UserManagerImpl @Inject constructor(
                     userEpisodeManager.removeCloudStatusFromFiles(playbackManager)
                 }
 
-                settings.setMarketingOptIn(false)
-                settings.setMarketingOptInNeedsSync(false)
+                settings.marketingOptIn.set(false)
+                settings.marketingOptIn.needsSync = false
```

Comment:
We can't just pass a false value for needsSync when we set the value here (i.e., `marketingOptIn.set(false, needsSync = false)` because that parameter only has effect when it is `true`. We have to explicitly set the needsSync value on its own. This certainly isn't ideal, and if this is something we end up wanting to do more often, I should probably rethink how this works, but my hope is that this isn't something we'll be doing apart from this one instance, so I left it as-is since I think the current handling makes more sense for the more common scenario where we don't want a call to set to unintentionally overwrite a previous request to sync a value.